modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java

```diff
@@ -6775,6 +6776,16 @@ else if (msg instanceof TcpDiscoveryJoinRequestMessage) {
                             TcpDiscoveryJoinRequestMessage req = (TcpDiscoveryJoinRequestMessage)msg;
 
                             if (!req.responded()) {
+                                if (sock instanceof SSLSocket &&
+                                    spi.ignite().configuration().isSslCertsTransmissionEnabled()) {
+                                    HashMap<String, Object> attrs = new HashMap<>(req.node().getAttributes());
+
+                                    attrs.put(ATTR_SECURITY_CERTIFICATES, ((SSLSocket) sock).getSession()
+                                        .getPeerCertificateChain());
```

Comment:
Default implementation (`X509V1CertImpl`) implements `Serializable`. We can add check here, and in case certificates aren't serializable - don't put them.